### PR TITLE
*: add a restriction for smooth-upgrade-tidb

### PR DESCRIPTION
**Pull request by zimulala**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: smooth-upgrade-tidb.md [hunk 1]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
     * 导入数据。
 
+    * 有分布式任务正在处理，或者升级中有分布式任务（即 [TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)和此功能还存在一些问题）。可以通过以下 3 种方法确保不触发此限制：
+      1. 没有开启过分布式执行框架（此功能默认关闭）。
+      2. 升级开始到升级完成前没有分布式任务（即没有 `ADD INDEX` 和 `IMPORT INTO` 操作）。
+      3. 可以通过 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 开关关闭分布式执行框架，且确认没有还在处理的分布式任务。
```
Comment (Frank945946) [inline smooth-upgrade-tidb.md:95]:
```suggestion
    * 已知缺陷，做平滑升级前，如果用户开启了 [TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)，且平滑升级时正通过 TiDB 分布式执行框架在执行 add index job 或者 import into job，可能会导致数据和索引不一致。可以通过以下几个步骤避免触发此问题：
      1. 将[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md) 参数设置为 ”OFF“（此功能默认关闭）。
      2. 升级开始前等待正在运行中的分布式 `ADD INDEX` 和 `IMPORT INTO` 操作完成，或者 Cancel 掉这些任务，等升级完成后再重新提交。
      3. 上述 2 个步骤完成后，再执行平滑升级。
```

File: smooth-upgrade-tidb.md [hunk 2]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
     * 导入数据。
 
+    * 已知缺陷，做平滑升级前，如果用户开启了 [TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)，且平滑升级时正通过 TiDB 分布式执行框架在执行 add index job 或者 import into job，可能会导致数据和索引不一致。可以通过以下几个步骤避免触发此问题：
```
Comment (zimulala) [inline smooth-upgrade-tidb.md:92]:
```suggestion
    * 做平滑升级前，如果用户开启了 [TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)，且平滑升级时正在执行 `ADD INDEX` 或者 `IMPORT INTO` 任务，可能会导致一些问题。可以通过以下步骤避免触发此问题：
```

File: smooth-upgrade-tidb.md [hunk 3]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
     * 导入数据。
 
+    * 做平滑升级前，如果用户开启了 [TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)，且平滑升级时正在执行 `ADD INDEX` 或者 `IMPORT INTO` 任务，可能会导致一些问题。可以通过以下步骤避免触发此问题：
+      1. 关闭 TiDB 分布式执行框架功能，即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为 ”OFF“（此为默认值）。
+      2. 升级开始前等待正在运行中的分布式 `ADD INDEX` 和 `IMPORT INTO` 操作完成，或者 Cancel 掉这些任务，等升级完成后再重新提交。
+      3. 上述 2 个步骤完成后，再执行平滑升级。
```
Comment (Oreoxmt) [inline smooth-upgrade-tidb.md:95]:
```suggestion
    * 如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，并且在升级过程中存在 `ADD INDEX` 或者 `IMPORT INTO` 任务，为避免可能发生的问题，请按以下步骤操作：
      1. 关闭 TiDB 分布式执行框架，即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为 `OFF`（默认值）。
      2. 在升级开始前，确保所有分布式 `ADD INDEX` 和 `IMPORT INTO` 任务已完成，或者取消这些任务并等待升级完成后重新开始。
      3. 完成上述两步后，开始进行平滑升级。
```

**Review by Frank945946 — APPROVED** [review]

File: smooth-upgrade-tidb.md [hunk 4]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
     * 导入数据。
 
+    * 如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，并且在升级过程中存在 `ADD INDEX` 或者 `IMPORT INTO` 任务，为避免可能发生的问题，请按以下步骤操作：
```
Comment (zimulala) [inline smooth-upgrade-tidb.md:92]:
```suggestion
    * 如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，并且在升级时有正在执行的 `ADD INDEX` 或者 `IMPORT INTO` 任务，为避免可能发生的问题，请按以下步骤操作：
```

**Review by Oreoxmt — APPROVED** [review]

File: smooth-upgrade-tidb.md [hunk 5]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
     * 导入数据。
 
+    * 如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，并且在升级时有正在执行的 `ADD INDEX` 或者 `IMPORT INTO` 任务，为避免可能发生的问题，请按以下步骤操作：
+      1. 关闭 TiDB 分布式执行框架，即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为 `OFF`（默认值）。
+      2. 在升级开始前，确保所有分布式 `ADD INDEX` 和 `IMPORT INTO` 任务已完成，或者取消这些任务并等待升级完成后重新开始。
+      3. 完成上述两步后，开始进行平滑升级。
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:95]: 这里的 2 和 3 属于升级前的检查项，要挪到 L80 后面吗

File: smooth-upgrade-tidb.md [hunk 6]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
     * 导入数据。
 
+    * 如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，并且在升级时有正在执行的 `ADD INDEX` 或者 `IMPORT INTO` 任务，为避免可能发生的问题，请按以下步骤操作：
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:92]:
```suggestion
    * 执行分布式 `ADD INDEX` 和 `IMPORT INTO` 任务。如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，并且在升级时有正在执行的 `ADD INDEX` 或者 `IMPORT INTO` 任务，为避免可能发生的问题，请按以下步骤操作：
```

File: smooth-upgrade-tidb.md [hunk 7]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
     * 导入数据。
 
+    * 执行分布式 `ADD INDEX` 或 `IMPORT INTO` 任务。如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，并且在升级时有正在执行的 `ADD INDEX` 或者 `IMPORT INTO` 任务，为避免可能发生的问题，请按以下步骤操作：
+      1. 关闭 TiDB 分布式执行框架，即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为 `OFF`（默认值）。
+      2. 在升级开始前，确保所有分布式 `ADD INDEX` 和 `IMPORT INTO` 任务已完成，或者取消这些任务并等待升级完成后重新开始。
+      3. 完成上述两步后，开始进行平滑升级。
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:95]:
```suggestion
    * 执行分布式 `ADD INDEX` 或 `IMPORT INTO` 任务。
```

File: smooth-upgrade-tidb.md [hunk 8]
```diff
@@ -89,6 +89,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
...
+
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:96]: please add the following after L81

```suggestion
如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，请关闭 TiDB 分布式执行框架（即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为默认值 `OFF`），并确保所有分布式 `ADD INDEX` 和 `IMPORT INTO` 任务已完成，或者取消这些任务并等待升级完成后重新开始。否则，升级期间的 `ADD INDEX` 操作，会导致数据和索引不一致。
```

File: smooth-upgrade-tidb.md (hunk 7 above)
Comment (qiancai) [inline smooth-upgrade-tidb.md:95]:
```suggestion
    * 执行分布式 `ADD INDEX` 或 `IMPORT INTO` 任务。
```

File: smooth-upgrade-tidb.md (hunk 8 above)
Comment (qiancai) [reply smooth-upgrade-tidb.md:96]: @ zimulala，PTAL

File: smooth-upgrade-tidb.md (hunk 7 above)
Comment (qiancai) [reply smooth-upgrade-tidb.md:95]: please check if we can move the following after L81

```
* 如果在平滑升级前，[TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，请关闭 TiDB 分布式执行框架（即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为默认值 `OFF`），并确保所有分布式 `ADD INDEX` 和 `IMPORT INTO` 任务已完成，或者取消这些任务并等待升级完成后重新开始。否则，升级期间的 `ADD INDEX` 操作，会导致数据和索引不一致。

```

Comment (zimulala) [reply smooth-upgrade-tidb.md:95]: 感觉可以写这里吧，主要是升级过程中存在这个，如果觉得不行也可以直接放到升级前也可以。

File: smooth-upgrade-tidb.md [hunk 9]
```diff
@@ -77,7 +77,11 @@ TiUP 会在 v1.14.0 版本自适应支持此功能，即无需特殊操作，直
 
 ### 用户操作限制
 
-* 在升级前，如果集群中存在正在处理的 canceling DDL job，即有正在被处理的 DDL job 被用户取消了，由于处于 canceling 状态的 job 无法被 `pause`，TiDB 会尝试重试。如果重试失败，会报错并退出升级。
+* 在升级前有如下两种限制：
+
+* 如果集群中存在正在处理的 canceling DDL job，即有正在被处理的 DDL job 被用户取消了，由于处于 canceling 状态的 job 无法被 `pause`，TiDB 会尝试重试。如果重试失败，会报错并退出升级。
+
+* 如果 [TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，请关闭 TiDB 分布式执行框架（即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为默认值 `OFF`），并确保所有分布式 `ADD INDEX` 和 `IMPORT INTO` 任务已完成，或者取消这些任务并等待升级完成后重新开始。否则，升级期间的 `ADD INDEX` 操作，可能导致数据索引不一致。
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:84]:
```suggestion
    * 如果集群中存在正在处理的 canceling DDL job，即有正在被处理的 DDL job 被用户取消了，由于处于 canceling 状态的 job 无法被 `pause`，TiDB 会尝试重试。如果重试失败，会报错并退出升级。

    * 如果 [TiDB 分布式执行框架](/tidb-distributed-execution-framework.md)已启用，请关闭 TiDB 分布式执行框架（即将 [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-从-v710-版本开始引入) 设置为默认值 `OFF`），并确保所有分布式 `ADD INDEX` 和 `IMPORT INTO` 任务已完成，或者取消这些任务并等待升级完成后重新开始。否则，升级期间的 `ADD INDEX` 操作可能导致数据索引不一致。
```

**Review by Oreoxmt — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Pull request merged**